### PR TITLE
Compose list support

### PR DESCRIPTION
**Pull request by manolovn**
### :goal_net: What's the goal?
Provide support for custom `ListRowItem`

https://user-images.githubusercontent.com/944814/142639667-d5e96cbd-4402-4149-8bd4-af4e326ddb53.mp4

https://user-images.githubusercontent.com/944814/142639646-7ffa53f5-2539-4afe-87f0-743559c7ed19.mp4

### ☑️ Checks
- [x] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
_If it cannot be tested explain why._
- [x] 🖼️ Screenshots/Videos
- [ ] Reviewed by Mistica design team

### CODE REVIEW

File: catalog-compose/build.gradle [hunk 1]
```diff
@@ -53,6 +53,8 @@ dependencies {
     implementation 'androidx.activity:activity-compose:1.3.1'
     implementation "com.google.android.material:compose-theme-adapter:1.0.2"
     implementation "androidx.navigation:navigation-compose:2.4.0-alpha10"
+    implementation "io.coil-kt:coil:1.4.0"
+    implementation "io.coil-kt:coil-compose:1.4.0"
```
Comment (jeslat) [inline catalog-compose/build.gradle:57]: Why do we need a library to load the images? Can't be done like previously without a library?

File: library-compose/src/main/java/com/telefonica/mistica/compose/list/Chip.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    color: Color = MisticaTheme.colors.badge,
+) {
+    Surface(
+        shape = RoundedCornerShape(4.dp),
+        color = color,
+    ) {
+        Text(
+            text = name.uppercase(),
+            style = MisticaTheme.typography.preset1Medium,
+            color = Color.White,
```
Comment (yamal-alm) [inline library-compose/src/main/java/com/telefonica/mistica/compose/list/Chip.kt:25]: were we using a hardcoded white before? shouldn't we use a Mistica constant? just asking..

**Review by yamal-alm — APPROVED** [review]
very good job @manolovn 👏

File: catalog-compose/build.gradle (hunk 1 above)
Comment (manolovn) [reply catalog-compose/build.gradle:57]: I've added it only for the catalog app, like a sample demo

File: library-compose/src/main/java/com/telefonica/mistica/compose/list/Chip.kt (hunk 2 above)
Comment (manolovn) [reply library-compose/src/main/java/com/telefonica/mistica/compose/list/Chip.kt:25]: uhm, I think it could be changed to textPrimaryInverse

**Review by jeslat — APPROVED** [review]
good job!

File: library-compose/src/main/java/com/telefonica/mistica/compose/list/ListRowItem.kt [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    val textColorSecondary = when (backgroundType) {
+        BackgroundType.TYPE_NORMAL,
+        BackgroundType.TYPE_BOXED,
+        -> MisticaTheme.colors.textSecondary
+        BackgroundType.TYPE_BOXED_INVERSE -> MisticaTheme.colors.textSecondaryInverse
+    }
+
+    Box(
+        modifier = boxModifier
+    ) {
```
Comment (gmerinojimenez) [inline library-compose/src/main/java/com/telefonica/mistica/compose/list/ListRowItem.kt:105]: just curious, why is this box needed?

File: library-compose/src/main/java/com/telefonica/mistica/compose/tag/Tag.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package com.telefonica.mistica.compose.tag
```
Comment (gmerinojimenez) [inline library-compose/src/main/java/com/telefonica/mistica/compose/tag/Tag.kt:1]: Same here, maybe (in another task) we can add a catalog page for this

**Review by gmerinojimenez — APPROVED** [review]
Amazing "currazo"

File: library-compose/src/main/java/com/telefonica/mistica/compose/list/ListRowItem.kt (hunk 3 above)
Comment (manolovn) [reply library-compose/src/main/java/com/telefonica/mistica/compose/list/ListRowItem.kt:105]: sure!

File: library-compose/src/main/java/com/telefonica/mistica/compose/tag/Tag.kt (hunk 4 above)
Comment (manolovn) [reply library-compose/src/main/java/com/telefonica/mistica/compose/tag/Tag.kt:1]: sure!

File: library-compose/src/main/java/com/telefonica/mistica/compose/badge/Badge.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package com.telefonica.mistica.compose.badge
```
[the remark this answers is not in the document]
Comment (manolovn) [reply library-compose/src/main/java/com/telefonica/mistica/compose/badge/Badge.kt:1]: sure!

File: library-compose/src/main/java/com/telefonica/mistica/compose/list/ListRowItem.kt (hunk 3 above)
Comment (manolovn) [reply library-compose/src/main/java/com/telefonica/mistica/compose/list/ListRowItem.kt:105]: I've added it to obtain the correct ripple effect for full-width rows and boxed ones

**Pull request merged**